src/backend/impl/src/repositories/types/proposal.rs

```diff
@@ -0,0 +1,63 @@
+use super::{DateTime, Uuid};
+use candid::{CandidType, Decode, Deserialize, Encode};
+use ic_stable_structures::{storable::Bound, Storable};
+use std::borrow::Cow;
+
+pub type ProposalId = Uuid;
+pub type NervousSystemProposalId = u64;
+pub type NeuronId = u64;
+
+#[derive(Debug, CandidType, Deserialize, Clone, PartialEq, Eq)]
+pub enum NervousSystem {
+    Network { id: NervousSystemProposalId },
```

Comment:
I've separated the NNS proposal id from our internal proposal id because when we'll add SNS proposals, there may be conflicts